### PR TITLE
adding mavlink setup for av_x1_v1 

### PR DESCRIPTION
**Pull request by AnnaDaiZH**
The previous mavlink setup for this board is generic and not specific for this board.

With this update the correct baud rate and mode is set and the required mavlink streams to run external vision in closed loop on a companion.

- HIGHRES_IMU provides the IMU data from the on board IMU to the Visual Inertial Odometry pipeline on the companion.

- CAMERA_TRIGGER provides timestamp information to enable the time synchronization between the onboard IMU and the captured images

- LOCAL_POSITION_NED provides ground truth from the local position estimation to the companion when fusing only GPS data in the EKF. By recording the mavros messages, the pose estimation performance of the VIO can be compared to the GPS positions offline. 

@dagar @DanielePettenuzzo

### CODE REVIEW

**Comment by dagar** [comment]
Should this be a mavlink preset for VIO? 
@mhkabir thoughts?

**Comment by mhkabir** [comment]
Yes this should be a preset for VIO. Have had the literally same patch in my VIO branch for several months - makes a lot of sense.

**Comment by DanielePettenuzzo** [comment]
I also think we need a new MAVLink mode for these kind of applications. This is really important for TX2 because with the max baud rate we can achieve the bandwidth is limited and we get jitter on imu data or triggers. I see a similar situation with obstacle avoidance. Moreover in many cases mavlink router is running on the companion computer so that you can also stream to qgc over wifi. Mavlink router doesn't support throttling down the streams therefore everything you stream to the companion computer also goes over wifi to the ground station. It would be nice to have a reduced mode (such as limited or osd) and then just add the necessary streams for the VIO or obstacle avoidance.

**Comment by AnnaDaiZH** [comment]
@DanielePettenuzzo  @dagar  cleaned up the rc.board and changed the serial port for TEL2 for av 

@mhkabir @dagar  I will make a new pr for a VIO mavlink preset called "MAVLINK_MODE_EV"

File: boards/av/x-v1/default.cmake [hunk 1]
```diff
@@ -12,7 +12,7 @@ px4_add_board(
 	SERIAL_PORTS
 		GPS1:/dev/ttyS6
 		TEL1:/dev/ttyS0
-		TEL2:/dev/ttyS1
+		TEL2:/dev/ttyS5
```
Comment (dagar) [inline boards/av/x-v1/default.cmake]: Could you explain the numbering change? Since the ports are unlabelled I was trying to keep the numbering consistent.

**Comment by DanielePettenuzzo** [comment]
@dagar the uart going to the tx2 is mapped to ttyS5 so it would be nice to have the same level of configuration as for other companion links? Is this a problem? Otherwise I think we would have to start it manually on ttyS5 in the rc.board for the av board

**Comment by AnnaDaiZH** [comment]
i rebased and updated the rc.board_extra.
Can you please close the pr before the structure changes again?
@dagar @DanielePettenuzzo

**Comment by dagar** [comment]
What's the status here? The PR now has submodule changes.

File: boards/av/x-v1/init/rc.board_extras [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #
 
 # AV-X: start MAVLink to companion (connected to TX2)
-mavlink start -d /dev/ttyS5 -r 20000 -b 460800 -m onboard
+mavlink start -d /dev/ttyS5 -b 500000 -m osd
```
Comment (dagar) [inline boards/av/x-v1/init/rc.board_extras]: Is this change intentional?

Comment (AnnaDaiZH) [reply boards/av/x-v1/init/rc.board_extras]: Yes, it is. With onboard mode, we had to many data drops, so changed it to osd

Comment (dagar) [reply boards/av/x-v1/init/rc.board_extras]: What about the baudrate? I thought 500000 was problematic for the tx2.

**Comment by dagar** [comment]
Continued in #11395
